src/libs/actions/IOU.js

```diff
@@ -99,26 +116,75 @@ function startLoadingAndResetError() {
 }
 
 /**
- * Creates IOUSplit Transaction
+ * Request money from another user
  *
- * @param {Object} params
- * @param {Number} params.amount
- * @param {String} params.comment
- * @param {String} params.currency
- * @param {String} params.debtorEmail
+ * @param {Object} report
+ * @param {Array} participants
+ * @param {Number} amount
+ * @param {String} currency
+ * @param {String} recipientEmail
+ * @param {String} debtorEmail
+ * @param {String} comment
  */
-function createIOUTransaction(params) {
-    startLoadingAndResetError();
-    DeprecatedAPI.CreateIOUTransaction(params)
-        .then((response) => {
-            if (response.jsonCode !== 200) {
-                processIOUErrorResponse(response);
-                return;
-            }
-
-            getIOUReportsForNewTransaction([response]);
-            Navigation.navigate(ROUTES.getReportRoute(response.chatReportID));
-        });
+function requestMoney(report, participants, amount, currency, recipientEmail, debtorEmail, comment) {
+    const chatReport = lodashGet(report, 'reportID', null) ? report : Report.buildOptimisticChatReport(participants);
+    const optimisticTransactionID = NumberUtils.rand64();
+    let iouReport;
+    if (chatReport.hasOutstandingIOU) {
+        iouReport = iouReports[`${ONYXKEYS.COLLECTION.REPORT_IOUS}${chatReport.iouReportID}`];
+        iouReport.total += amount;
+    } else {
+        iouReport = ReportUtils.buildOptimisticIOUReport(recipientEmail, debtorEmail, amount, chatReport.reportID, currency, 'en');
+    }
+    const optimisticReportAction = ReportUtils.buildOptimisticIOUReportAction('create', amount, comment, currency, '', optimisticTransactionID, iouReport.reportID, debtorEmail);
+    const optimisticData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${chatReport.reportID}`,
+            value: {
+                [optimisticReportAction.reportActionID]: {
```

Comment:
Why are we introducing `reportActionID` as a key in this data structure? We don't use it anywhere else for any reason. All of the logic in the app assumes that the keys will be either a `clientID` or a `sequenceNumber`.

Reply:
Yep, the backend will actually clean this up and use the sequence number, but I don't think we can safely assume the sequence number here while offline, which is why we are using the reportActionID as an easy-to-identify random number.
I'm not familiar with the clientID use, though! Is that a better option that could fit our case?

Reply:
At the moment, we use a `clientID` as the key. The `sequenceNumber` we "guess", but in your PR it looks like `sequenceNumber` and `clientID` are the same (I think this is probably wrong and also needs to be updated).

https://github.com/Expensify/App/blob/95161b5962fbd03a72c3fcd0e3c83352b3c60d46/src/libs/ReportUtils.js#L654-L657

The `sequenceNumber` should not be set to the `clientID` it should be an estimated `sequenceNumber` like this:

https://github.com/Expensify/App/blob/a24de554bd2e70575f4389942d6132d7ca9ea082/src/libs/actions/Report.js#L809-L817

Otherwise we won't be able to mark the comments as "unread".